### PR TITLE
Update documentation and readme

### PR DESCRIPTION
**Pull request by andrewmc2**
- Update clang format
- Wrote header file documentation for TWGNotificationObserver
- Changed all NSArray properties and parameters to only accept arrays containing NSString objects
- Updated podspec author info
- Removed unneeded info in test file
- Gave credit to TWG where needed
- Update readme
- Update version

### CODE REVIEW

File: Example/Pods/Target Support Files/Pods-TWGNotificationObserver_Tests/Pods-TWGNotificationObserver_Tests-acknowledgements.plist [hunk 1]
```diff
@@ -50,38 +50,38 @@ THE SOFTWARE.
...
       and conversions to other media types.
 
-      "Work" shall mean the work of authorship, whether in Source or
+      &quot;Work&quot; shall mean the work of authorship, whether in Source or
       Object form, made available under the License, as indicated by a
       copyright notice that is included in or attached to the work
       (an example is provided in the Appendix below).
 
-      "Derivative Works" shall mean any work, whether in Source or Object
+      &quot;Derivative Works&quot; shall mean any work, whether in Source or Object
```
Comment (bgilham) [inline Example/Pods/Target Support Files/Pods-TWGNotificationObserver_Tests/Pods-TWGNotificationObserver_Tests-acknowledgements.plist]: Any clue why all of the quotes were changed?

File: Example/Tests/TWGNotificationObserverTests.m [hunk 2]
```diff
@@ -21,102 +13,90 @@ - (void)deregisterObservers;
...
 
 @end
 
 @implementation TWGNotificationObserverTests
 
-- (void)setUp {
-  [super setUp];
+- (void)setUp
+{
+    [super setUp];
```
Comment (bgilham) [inline Example/Tests/TWGNotificationObserverTests.m]: Are we missing setting the notificationObserver up here? If we don't need `setUp`, we should remove it.

File: TWGNotificationObserver.podspec [hunk 3]
```diff
@@ -8,7 +8,7 @@
 
 Pod::Spec.new do |s|
   s.name             = "TWGNotificationObserver"
-  s.version          = "0.1.0"
+  s.version          = "0.1.1"
   s.summary          = "Helper for working with NSNotificationCenter"
 
 # This description is used to generate tags and improve search results.
```
Comment (bgilham) [inline TWGNotificationObserver.podspec]: Let's remove all of the commented lines in the pod spec too.

**Comment by bgilham** [comment]
@andrewmc2 Just some really quick notes. Great work, man :+1:

File: Example/Pods/Target Support Files/Pods-TWGNotificationObserver_Tests/Pods-TWGNotificationObserver_Tests-acknowledgements.plist (hunk 1 above)
Comment (andrewmc2) [inline Example/Pods/Target Support Files/Pods-TWGNotificationObserver_Tests/Pods-TWGNotificationObserver_Tests-acknowledgements.plist]: I have no idea - I didn't make those changes manually

**Comment by bgilham** [comment]
Good job @andrewmc2 :+1:

**Pull request merged**